### PR TITLE
ci: prevent concurrent tests

### PR DESCRIPTION
**Pull request by millsp**
Turns out this was already enabled! That said, it was not enabled for benchmarks, so I activated it there. We also had some old cancellation logic that I ported to the newer native GH way of doing it.

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -37,6 +37,10 @@ on:
 env:
   PRISMA_TELEMETRY_INFORMATION: 'prisma benchmark.yml'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```
Comment (millsp) [inline .github/workflows/benchmark.yml:42]: https://docs.github.com/en/actions/using-jobs/using-concurrency#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -53,6 +53,10 @@ env:
   # To hide "Update available 0.0.0 -> x.x.x"
   PRISMA_HIDE_UPDATE_MESSAGE: true
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.event_name != 'schedule' }}
```
Comment (millsp) [inline .github/workflows/test.yml:58]: https://docs.github.com/en/actions/using-jobs/using-concurrency#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow

Comment (millsp) [reply .github/workflows/test.yml:58]: `${{ github.event_name != 'schedule' }}` so that we don't cancel on scheduled runs (for BuildPulse)

**Review by aqrln — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (Jolg42) [inline .github/workflows/test.yml:58]:
```suggestion
  // Don't cancel if it's on schedule (used for BuildPulse for flaky tests detection)
  cancel-in-progress: ${{ github.event_name != 'schedule' }}
```

**Review by Jolg42 — APPROVED** [review]

**Review by jkomyno — APPROVED** [review]

**Pull request merged**